ajax/viewsubitem.php

```diff
@@ -51,10 +51,13 @@
     && ($parent = getItemForItemtype($_POST['parenttype']))
 ) {
     if (
-        isset($_POST[$parent->getForeignKeyField()])
+        isset($_POST[$parent::getForeignKeyField()])
         && isset($_POST["id"])
-        && $parent->getFromDB($_POST[$parent->getForeignKeyField()])
+        && $parent->getFromDB($_POST[$parent::getForeignKeyField()])
     ) {
+        if (!$item::isNewID($_POST['id'])) {
+            $item->getFromDB($_POST["id"]);
+        }
```

Comment:
grep -rni src -e 'showForm([.]*\$id,' | wc -l
> 98

grep -rnis src -e 'showForm([.]*\$id,' -A 10 | grep 'this->getFromDB' | wc -l
> 7

This seems the opposite of how it is actually handled. Classes don't have any option to *not* have an ID parameter because the signature in CommonDBTM has $ID as the first parameter.
From what I can see, only approximately 7% of classes that implement this method will potentially mutate itself by calling `getFromDB`. Some use the `ID` for permission checks even though they use the fields already loaded in the object instance and it could be a different item. The default implementation uses the `ID` parameter for permission checks and to check if it is a new item and then passes the current object instance to the generic form twig template without any reloading of fields.

Having a `show*` method potentially change the item the object is referring to seems like an unwanted behavior most of the time.

Reply:
```grep -rni src -e 'showForm([.]*$id,' | wc -l```
98

```grep -rnis src -e 'showForm([.]*$id,' -A 10 | grep 'this->getFromDB' | wc -l```
7

This seems the opposite of how it is actually handled. Classes don't have any option to _not_ have an ID parameter because the signature in CommonDBTM has $ID as the first parameter. From what I can see, only approximately 7% of classes that implement this method will potentially mutate itself by calling `getFromDB`. Some use the `ID` for permission checks even though they use the fields already loaded in the object instance and it could be a different item. The default implementation uses the `ID` parameter for permission checks and to check if it is a new item and then passes the current object instance to the generic form twig template without any reloading of fields.
> 
> Having a `show*` method potentially change the item the object is referring to seems like an unwanted behavior most of the time.

Reply:
`CommonDBTM::can()` is (re)loading the item most of the time, I guess.

I still think it should be better to (re)load the item in the `showForm()` method, to prevent unexpected issues later.